File: CodingInterviewsJavaSolutions/708. Insert into a Sorted Circular Linked List.java
class Solution {
    public Node insert(Node head, int insertVal) {
        if (head == null) {
            Node new_node = new Node(insertVal);
            new_node.next = new_node;
            return new_node;
        }
        
        Node cur = head;
        while (true) {
            int first = cur.val;
            int second = cur.next.val;
            
            if (first <= insertVal && insertVal <= second) {
                break;
            }
            
            if (first > second) {
                if (first <= insertVal || insertVal <= second) {
                    break;
                }
            }
            
            cur = cur.next;
            if (cur == head) {
                break;
            }
        }
        
        Node new_node = new Node(insertVal, cur.next);
        cur.next = new_node;
        return head;
    }
}

File: CodingInterviewsPythonSolutions/708. Insert into a Sorted Circular Linked List.py
# Definition for a Node.
class Node:
    def __init__(self, val=None, next=None):
        self.val = val
        self.next = next

class Solution:
    def insert(self, head: 'Optional[Node]', insertVal: int) -> 'Node':
        if not head:
            new_node = Node(insertVal)
            new_node.next = new_node
            return new_node
        
        cur = head
        while True:
            first = cur.val
            second = cur.next.val
            
            if first <= insertVal <= second:
                break
            
            if first > second:
                if first <= insertVal or insertVal <= second:
                    break
            
            cur = cur.next
            if cur == head:
                break
        
        new_node = Node(insertVal, cur.next)
        cur.next = new_node
        return head
